data.yaml

```diff
@@ -1,31 +1,31 @@
 c:
-  cycle: "17.0"
-  releaseDate: "2018-06-01"
+  cycle: "9.5"
+  releaseDate: "2022-05-27"
```

Comment:
From where did you get this release data @glokta1 ?
I was using [wikipedia](https://en.wikipedia.org/wiki/C_(programming_language)) for this btw, we want to be completely sure about this release cycle data.

Reply:
Tagging you again, @glokta1 in case you missed this.

Reply:
@ryan-gang In naming the Dockerfiles  for C, we're using the GCC compiler version (`c-9.2.Dockerfile`) so I thought to make it consistent here. Used this: https://gcc.gnu.org/gcc-9/

This is not ideal of course. We should do the same thing we're doing for C++ where we mention the standard, not the associated compiler version. 

cc: @rohitpaulk

Reply:
Ah. Yep we should change this, but wouldn't be backward-compatible - let's add to Linear to think & figure out how to best do this. For the current usage here, let's stick to whatever makes most sense in the meantime

Reply:
@rohitpaulk Created a Linear issue here: https://linear.app/codecrafters/issue/CC-1314/update-naming-of-c-dockerfile

Merging this as is for now.